MC_data_new.py

```diff
@@ -166,439 +236,369 @@ def SaveEventInfo(info_dict, folder, out):  # THIS FUNCTION IS CURRENTLY NOT USE
+                param_tree.Branch('y_vertex',y_vertex,"y_vertex/F")
+                param_tree.Branch('z_vertex',z_vertex,"z_vertex/F")
+                param_tree.Branch('x_vertex_end',x_vertex_end,"x_vertex_end/F")
+                param_tree.Branch('y_vertex_end',y_vertex_end,"y_vertex_end/F")
+                param_tree.Branch('z_vertex_end',z_vertex_end,"z_vertex_end/F")
+                param_tree.Branch('energy',energy,"energy/F")
+                param_tree.Branch('px',px,"px/F")
+                param_tree.Branch('py',py,"py/F")
+                param_tree.Branch('pz',pz,"pz/F")
+                param_tree.Branch('theta',theta,"theta/F")
+                param_tree.Branch('phi',phi,"phi/F") 
+                param_tree.Branch('nhits_og',nhits_og,"nhits_og/I")
+                param_tree.Branch('xhits_og',xhits_og,"xhits_og[nhits_og]/F")
+                param_tree.Branch('yhits_og',yhits_og,"yhits_og[nhits_og]/F")
+                param_tree.Branch('zhits_og',zhits_og,"zhits_og[nhits_og]/F")
+                param_tree.Branch('EDepHit_og',EDepHit_og,"EDepHit_og[nhits_og]/F")  
+
+
+                final_imgs=list();
                 
-                    infilename=infile[:-5]    
-                    #outfile=rt.TFile('%s/histograms_Run%05d.root' % (opt.outfolder,run_count), 'RECREATE') #OUTPUT NAME (only run number)
-                    #outfile=rt.TFile('%s/histograms_%s_%d_mm_%d_V.root' % (opt.outfolder, infilename, opt.z_gem-z_ini, opt.GEM1_HV), 'RECREATE') #OUTPUT NAME WITH PARAMETERS INFO
-                    outfile=rt.TFile('%s/histograms_%s_%d_mm_%d_V_%s.root' % (opt.outfolder, infilename, opt.z_gem-z_ini, opt.GEM1_HV, time.strftime("%Y%m%d-%H%M%S") ), 'RECREATE') #OUTPUT NAME WITH PARAMETERS INFO AND TIMESTAMP
-                    outfile.mkdir('event_info')
-                    SaveValues(params, outfile) ## SAVE PARAMETERS OF THE RUN
-                    outtree = rt.TTree("info_tree", "info_tree")
-                    outtree.Branch("eventnumber", eventnumber, "eventnumber/I")
-                    outtree.Branch("particle_type", particle_type, "particle_type/I")
-                    outtree.Branch("energy_ini", energy_ini, "energy_ini/F")
-                    outtree.Branch("theta_ini", theta_ini, "theta_ini/F")
-                    outtree.Branch("phi_ini", phi_ini, "phi_ini/F")
-
-                    final_imgs=list();
+                if opt.events==-1:
+                    totev=tree.GetEntries()
+                else:
                     
-                    if opt.events==-1:
-                        totev=tree.GetEntries()
+                    if opt.events<=tree.GetEntries():
+                        totev=opt.events
                     else:
+                        totev=tree.GetEntries()
+                    
+                VignMap=rt.TH2D()
+                if(opt.Vignetting):
+                    VignFile=rt.TFile.Open('VignettingMap/'+opt.Vig_Map,'read')
```

Comment:
@SamueleTorelli I am trying to run the code. Where can I find the vignetting map you're using here? Also, I see the parameter `Vig_Map` is only in `ConfigFile_MAN.txt`. So we remove `ConfigFile_new.txt, right`? Thanks

Reply:
Sorry, they are in my repository, they were not added to git. By the way, ConfigFile_MAN.txt is for MANGO, while ConfigFile_new.txt is for lime, you should use the second, adding the vignetting map information. The maps are now on my repository, which I will merge after this commit is accepted